### ISSUE TITLE
Add possibility to return promise instead of array in request event

### ISSUE THREAD

**Issue by ghost** [werk85/fetch-intercept#23]
My scenario is as follows:
1. A fetch() call is made with a GET request to a REST API
2. The 'request' event is handled so that it injects an auth token to the headers
3. If the token is expired I'd like to refresh the token by making an additional call inside the 'request' event

The question is as follows: would it be possible to return a Promise from the 'request' event instead of a fixed array with the [url, config]? The token refresh call is made using a promise, and I'd like to inject the auth header in the .then of the promise. The 'request' does not wait for the promise resolve. Example code of what' I'm trying to do:
```
this.unregisterFetchInterceptor = fetchInterceptor.register({
         request: (url, config) => {
               this.refreshAuthTokenIfExpired(resourceName).then(() => {
                     let modifiedConfig = this.injectAuthHeader(resourceName, config);
                     return [url, modifiedConfig];
               });                   
         }
 });
```
Thanks,
Adam

**Comment by mlegenhausen** [comment]
Make sure to return the promise from the `request` function.

I think you fixed your typing problems in #24? @aluczak can this issue be closed?

**Comment by ghost** [comment]
Yes. Closing now.

Cheers.
